patacrep/build.py

```diff
@@ -84,42 +85,60 @@ def write_tex(self, output):
         - output: a file object, in which the file will be written.
         """
         # Updating configuration
-        config = DEFAULT_CONFIG.copy()
-        config.update(self.config)
+        self._config = DEFAULT_CONFIG.copy()
+        self._config.update(self._raw_config)
         renderer = TexBookRenderer(
-            config['template'],
-            config['datadir'],
-            config['lang'],
-            config['encoding'],
+            self._config['template'],
+            self._config['datadir'],
+            self._config['lang'],
+            self._config['encoding'],
             )
-        config.update(renderer.get_variables())
-        config.update(self.config)
+        self._config.update(renderer.get_variables())
+        self._config.update(self._raw_config)
 
-        config['_compiled_authwords'] = authors.compile_authwords(
-            copy.deepcopy(config['authwords'])
+        self._config['_compiled_authwords'] = authors.compile_authwords(
+            copy.deepcopy(self._config['authwords'])
             )
 
         # Loading custom plugins
-        config['_content_plugins'] = files.load_plugins(
-            datadirs=config.get('datadir', []),
+        self._config['_content_plugins'] = files.load_plugins(
+            datadirs=self._config.get('datadir', []),
             root_modules=['content'],
             keyword='CONTENT_PLUGINS',
             )
-        config['_song_plugins'] = files.load_plugins(
-            datadirs=config.get('datadir', []),
+        self._config['_song_plugins'] = files.load_plugins(
+            datadirs=self._config.get('datadir', []),
             root_modules=['songs'],
             keyword='SONG_RENDERERS',
             )['tsg']
 
         # Configuration set
-        config['render'] = content.render
-        config['content'] = content.process_content(
-            config.get('content', []),
-            config,
+        self._config['render'] = content.render
+        self._config['content'] = content.process_content(
+            self._config.get('content', []),
+            self._config,
             )
-        config['filename'] = output.name[:-4]
+        self._config['filename'] = output.name[:-4]
 
-        renderer.render_tex(output, config)
+        renderer.render_tex(output, self._config)
+
+    def has_errors(self):
+        """Return `True` iff errors have been encountered in the book.
+
+        Note that `foo.has_errors() == False` does not means that the book has
+        not any errors: it does only mean that no error has been found yet.
+        """
+        for _ in self.iter_errors():
```

Comment:
Je n'ai quasiment pas testé, mais normalement, non : chaque appel à `self.iter_errors()` crée un *nouvel* itérateur, donc ça recommence à zéro à chaque fois.